messaging/WFLY-9213_Implement_the_Pause_method_for_a_Topic.adoc

```diff
@@ -0,0 +1,102 @@
+= Implement the Pause method for a Topic via Management APIs
+:author:            Emmanuel Hugonnet
+:email:             ehugonne@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:          messaging,jms
+:idseparator:       -
+
+== Overview
+
+
+An Artemis Queue can be paused but there is nothing for a Topic. The goal of this feature is to be able to pause a Topic by pausing all its consumers . New subscriptions being registered while the topic is in pause will be paused also.
+This is implemented in Apache Artemis and we need to expose this feature in the WildFly management API.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-9213[WFLY-9213]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/ENTMQBR-1002[ENTMQBR-1002]
+* https://issues.apache.org/jira/browse/ARTEMIS-2401[ARTEMIS-2401]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Implementation Plan
+
+
+Adding a `pause` and a `resume` methods on the resource http://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/server/jms-topic/index.html[/subsystem=messaging-activemq/server=default/jms-topic].
+The `pause` method has a boolean `persist` parameter which is optionnal and default to false.
+We need also to add a runtime only boolean `paused`attribute to http://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/server/jms-topic/index.html[/subsystem=messaging-activemq/server=default/jms-topic] to indicate the status of the Topic.
+
+CLI to execute the commands:
+
+----
+/subsystem=messaging-activemq/server=default/jms-topic=topic:pause()
+{
+    "outcome" => "success",
+    "result" => undefined
+}
+
+/subsystem=messaging-activemq/server=default/jms-topic=topic:read-attribute(name=paused)
+{
+    "outcome" => "success",
+    "result" => true
+}
+
+/subsystem=messaging-activemq/server=default/jms-topic=topic:resume()
+{
+    "outcome" => "success",
+    "result" => undefined
+}
+
+/subsystem=messaging-activemq/server=default/jms-topic=topic:read-attribute(name=paused)
+{
+    "outcome" => "success",
+    "result" => false
+}
+----
+
+
+== Test Plan
+
+== Community Documentation
+
+* Covered by operation description in management model.
```

Comment:
If it will cover the effect of having topic "paused" I see no problem here.

Reply:
Maybe we can also put a link towards Artemis documentation on this

Reply:
+1, though this would mean we would have to maintain the link to target correct destination...